libraries/helpers.rb

```diff
@@ -160,6 +160,39 @@ def ami_bootstrapped?
   version != '' && ( node['cfncluster']['skip_install_recipes'] == 'yes' || node['cfncluster']['skip_install_recipes'] == true )
 end
 
+#
+# Check if OS type specified by the user is the same as the OS identified by Ohai
+#
+def validate_OS_type
+  case node['platform']
+  when 'ubuntu'
+    current_os = 'ubuntu'+ node['platform_version'].tr('.','')
```

Comment:
Can we think about having a common function to retrieve OS from ohai, and use it everywhere in the cookbook? For example we currently have scattered logic [like this](https://github.com/aws/aws-parallelcluster-cookbook/blob/v2.9.1/attributes/default.rb#L106) and in other places when we have to determine the OS in cookbook.
Is it possible to use a common logic for all of these so we don't have to keep track of these logic separately?

Reply:
Can I do it in a separate PR? This change requires change in multiple files that are not relevant to the current task. And separating the PR also helps to identify the reason if one of the PR fails some tests.